### ISSUE TITLE
where is the val code in u2net_train?

### ISSUE THREAD

**Issue by shantzhou** [xuebinqin/U-2-Net#245]
hello, I couldn't find the val part in u2net_train,  how to know the quantity of the model without val。

**Comment by xuebinqin** [comment]
Hi, Zhou, Thanks for your interests. Please refer to the detailed training settings in our paper. For the training of the SOD dataset, we followed other papers without using the validation process. What we did before is plotting the -log(loss) of the training process. If the training curve is flat, it should be ok to stop (In the model development process, we actually validate and evaluated all the hundreds of intermediate model weights of many different models, we found there was no obvious overfitting in terms of the F-measure. So don't worry about that if you are training on DUTS-TR and test on the rest datasets mentioned in our paper. We guess that's because all the test datasets are from different groups at different time, whose differences are significant enough.). In your case, if you want to get better
results on SOD task, you can also pick one of the other SOD datasets, e.g. MSRA and so on, as your validation set. Then select the best model based on the validation metrics such as F-measure, MAE, etc. We also gave our suggested iteration numbers (around 400K on DUTS-TR dataset) in the paper. All the best.

**Comment by shantzhou** [comment]
oh, thanks for your reply， I will study your paper.
best wishes!&nbsp;


------------------&nbsp;原始邮件&nbsp;------------------
发件人:                                                                                                                        "xuebinqin/U-2-Net"                                                                                    ***@***.***&gt;;
发送时间:&nbsp;2021年8月17日(星期二) 下午4:37
***@***.***&gt;;
抄送:&nbsp;"Shente ***@***.******@***.***&gt;;
主题:&nbsp;Re: [xuebinqin/U-2-Net] where is the val code in u2net_train? (#245)





 
Hi, Zhou, Thanks for your interests. Please refer to the detailed training settings in our paper. For the training of the SOD dataset, we followed other papers without using the validation process. What we did before is plotting the -log(loss) of the training process. If the training curve is flat, it should be ok to stop (In the model development process, we actually validate and evaluated all the hundreds of intermediate model weights of many different models, we found there was no obvious overfitting in terms of the F-measure. So don't worry about that if you are training on DUTS-TR and test on the rest datasets mentioned in our paper. We guess that's because all the test datasets are from different groups at different time, whose differences are significant enough.). In your case, if you want to get better
 results on SOD task, you can also pick one of the other SOD datasets, e.g. MSRA and so on, as your validation set. Then select the best model based on the validation metrics such as F-measure, MAE, etc. We also gave our suggested iteration numbers (around 400K on DUTS-TR dataset) in the paper. All the best.
 
—
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub, or unsubscribe.
Triage notifications on the go with GitHub Mobile for iOS or Android.
